osu.Game.Rulesets.Taiko/UI/TaikoPlayfield.cs

```diff
@@ -255,16 +267,30 @@ internal void OnNewResult(DrawableHitObject judgedObject, JudgementResult result
 
                     break;
             }
-        }
 
-        private class ProxyContainer : LifetimeManagementContainer
-        {
-            public new MarginPadding Padding
+            if (characterDrawable.Drawable is DrawableTaikoMascot mascot)
             {
-                set => base.Padding = value;
+                var isFailing = result.Type == HitResult.Miss;
+
+                // Only take combo in consideration when it's not a strong hit (it's always false)
+                if (!(judgedObject.HitObject is StrongHitObject))
+                {
+                    if (isFailing)
+                        isFailing = result.Judgement.AffectsCombo;
+                }
+
+                mascot.PlayfieldState.Value = isFailing ? TaikoMascotAnimationState.Fail : TaikoMascotAnimationState.Idle;
             }
+        }
+    }
 
-            public void Add(Drawable proxy) => AddInternal(proxy);
+    internal class ProxyContainer : LifetimeManagementContainer
```

Comment:
why did this become `internal`? don't see any other new references to this class in this PR.

Reply:
My analyzer script was outdated or something, I'll revert this line back.